cl/lib/search_utils.py

```diff
@@ -261,3 +318,428 @@ def store_search_api_query(
+        "search_form": search_form,
+        "search_summary_str": search_summary_str,
+        "search_summary_dict": search_summary_dict,
+        "error": error,
+        "courts": courts,
+        "court_count_human": court_count_human,
+        "court_count": court_count,
+        "query_citation": query_citation,
+        "cited_cluster": cited_cluster,
+        "related_cluster": related_cluster,
+        "facet_fields": facet_fields,
+        "error_message": error_message,
+        "suggested_query": suggested_query,
+        "estimated_count_threshold": simplify_estimated_count(
+            compute_lowest_possible_estimate(
+                settings.ELASTICSEARCH_CARDINALITY_PRECISION
+            )
+        ),
+        "missing_citations": missing_citations_str,
+    }
+
+
+def get_headers_and_transformations_for_search_export(
+    type: str,
+) -> tuple[list[str], dict[str, Callable[..., Any]]]:
+    """
+    Retrieves CSV headers and data transformation functions for a given search
+    type.
+
+    This function determines the appropriate CSV headers and data transformation
+    functions based on the specified Elasticsearch search type. It combines
+    headers and transformations from relevant document classes to generate a
+    comprehensive set for CSV export.
+
+    :param type: The type of Elasticsearch search to be performed. Valid values
+        are defined in the `SEARCH_TYPES` enum.
+    :return:  A tuple containing:
+        - A list of strings representing the CSV headers.
+        - A dictionary where keys are field names and values are callable functions
+          that define the data transformations.
+    """
+    match type:
+        case SEARCH_TYPES.PEOPLE:
+            keys = PersonDocument.get_csv_headers()
+            transformations = PersonDocument.get_csv_transformations()
+        case SEARCH_TYPES.ORAL_ARGUMENT:
+            keys = AudioDocument.get_csv_headers()
+            transformations = AudioDocument.get_csv_transformations()
+        case SEARCH_TYPES.PARENTHETICAL:
+            keys = ParentheticalGroupDocument.get_csv_headers()
+            transformations = (
+                ParentheticalGroupDocument.get_csv_transformations()
+            )
+        case SEARCH_TYPES.RECAP | SEARCH_TYPES.DOCKETS:
+            keys = (
+                DocketDocument.get_csv_headers()
+                + ESRECAPDocument.get_csv_headers()
+            )
+            transformations = (
+                DocketDocument.get_csv_transformations()
+                | ESRECAPDocument.get_csv_transformations()
+            )
+        case SEARCH_TYPES.OPINION:
+            keys = (
+                OpinionClusterDocument.get_csv_headers()
+                + OpinionDocument.get_csv_headers()
+            )
+            transformations = (
+                OpinionClusterDocument.get_csv_transformations()
+                | OpinionDocument.get_csv_transformations()
+            )
+
+    return keys, transformations
```

Comment:
It seems that these variables can be referenced before assignment. Can we just return default empty values for them?

Reply:
Sure!